kernel: bump 5.15 to 5.15.86

graysky2 opened the pull request:
Removed upstreamed:
```
  pending-5.15/101-Use-stddefs.h-instead-of-compiler.h.patch[1]
  ipq806x/patches-5.15/122-01-clk-qcom-clk-krait-fix-wrong-div2-functions.patch[2]
  bcm27xx/patches-5.15/950-0198-drm-fourcc-Add-packed-10bit-YUV-4-2-0-format.patch[3]
```
Manually rebased:
```
  ramips/patches-5.15/100-PCI-mt7621-Add-MediaTek-MT7621-PCIe-host-controller-.patch[4]
```

Added patch/backported:
```
  ramips/patches-5.15/107-PCI-mt7621-Add-sentinel-to-quirks-table.patch[5]
```

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.86&id=c160505c9b574b346031fdf2c649d19e7939ca11
2. Cannot find in the stable tree but it is here: https://github.com/torvalds/linux/commit/a051e10bfc6906d29dae7a31f0773f2702edfe1b
3. https://github.com/gregkh/linux/commit/ec1727f89ecd6f2252c0c75e200058819f7ce47a
4. Quilt gave this output when I applied the patch to rebase it:
```
% quilt push -f
Applying patch platform/100-PCI-mt7621-Add-MediaTek-MT7621-PCIe-host-controller-.patch patching file arch/mips/ralink/Kconfig
patching file drivers/pci/controller/Kconfig
patching file drivers/pci/controller/Makefile
patching file drivers/staging/Kconfig
patching file drivers/staging/Makefile
patching file drivers/staging/mt7621-pci/Kconfig
patching file drivers/staging/mt7621-pci/Makefile
patching file drivers/staging/mt7621-pci/TODO
patching file drivers/staging/mt7621-pci/mediatek,mt7621-pci.txt patching file drivers/staging/mt7621-pci/pci-mt7621.c Hunk #1 FAILED at 1.
Not deleting file drivers/staging/mt7621-pci/pci-mt7621.c as content differs from patch 1 out of 1 hunk FAILED -- saving rejects to file drivers/staging/mt7621-pci/pci-mt7621.c.rej patching file drivers/pci/controller/pcie-mt7621.c Applied patch platform/100-PCI-mt7621-Add-MediaTek-MT7621-PCIe-host-controller-.patch (forced; needs refresh)
```

Upon inspecting `drivers/staging/mt7621-pci/pci-mt7621.c.rej`, it seems that the original patch wants to delete `drivers/staging/mt7621-pci/pci-mt7621.c` but upstream's version was not an exact match.  I opted to delete that file and need some feedback.  Was that the correct course of action?

5. Suggestion by hauke: https://github.com/torvalds/linux/commit/19098934f910b4d47cb30251dd39ffa57bef9523
"This patch is in upstream kernel, but it was backported to the old staging driver in kernel 5.15."

Build system: x86_64
Build-tested: bcm2711/RPi4B, filogic/xiaomi_redmi-router-ax6000-ubootmod
Run-tested: bcm2711/RPi4B, filogic/xiaomi_redmi-router-ax6000-ubootmod

Signed-off-by: John Audia <therealgraysky@proton.me>

graysky2 committed:
kernel: bump 5.15 to 5.15.86

Removed upstreamed:
  pending-5.15/101-Use-stddefs.h-instead-of-compiler.h.patch[1]
  ipq806x/patches-5.15/122-01-clk-qcom-clk-krait-fix-wrong-div2-functions.patch[2]
  bcm27xx/patches-5.15/950-0198-drm-fourcc-Add-packed-10bit-YUV-4-2-0-format.patch[3]

Manually rebased:
  ramips/patches-5.15/100-PCI-mt7621-Add-MediaTek-MT7621-PCIe-host-controller-.patch[4]

Added patch/backported:
  ramips/patches-5.15/107-PCI-mt7621-Add-sentinel-to-quirks-table.patch[5]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.86&id=c160505c9b574b346031fdf2c649d19e7939ca11
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.86&id=a051e10bfc6906d29dae7a31f0773f2702edfe1b
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.86&id=ec1727f89ecd6f2252c0c75e200058819f7ce47a
4. Quilt gave this output when I applied the patch to rebase it:
% quilt push -f
Applying patch platform/100-PCI-mt7621-Add-MediaTek-MT7621-PCIe-host-controller-.patch
patching file arch/mips/ralink/Kconfig
patching file drivers/pci/controller/Kconfig
patching file drivers/pci/controller/Makefile
patching file drivers/staging/Kconfig
patching file drivers/staging/Makefile
patching file drivers/staging/mt7621-pci/Kconfig
patching file drivers/staging/mt7621-pci/Makefile
patching file drivers/staging/mt7621-pci/TODO
patching file drivers/staging/mt7621-pci/mediatek,mt7621-pci.txt
patching file drivers/staging/mt7621-pci/pci-mt7621.c
Hunk #1 FAILED at 1.
Not deleting file drivers/staging/mt7621-pci/pci-mt7621.c as content differs from patch
1 out of 1 hunk FAILED -- saving rejects to file drivers/staging/mt7621-pci/pci-mt7621.c.rej
patching file drivers/pci/controller/pcie-mt7621.c
Applied patch platform/100-PCI-mt7621-Add-MediaTek-MT7621-PCIe-host-controller-.patch (forced; needs refresh)

Upon inspecting drivers/staging/mt7621-pci/pci-mt7621.c.rej, it seems that
the original patch wants to delete drivers/staging/mt7621-pci/pci-mt7621.c
but upstream's version was not an exact match.  I opted to delete that
file.

5. Suggestion by hauke: https://github.com/torvalds/linux/commit/19098934f910b4d47cb30251dd39ffa57bef9523
"This patch is in upstream kernel, but it was backported to the old
staging driver in kernel 5.15."

Build system: x86_64
Build-tested: bcm2711/RPi4B, filogic/xiaomi_redmi-router-ax6000-ubootmod
Run-tested: bcm2711/RPi4B, filogic/xiaomi_redmi-router-ax6000-ubootmod

Signed-off-by: John Audia <therealgraysky@proton.me>

hauke commented on target/linux/bcm27xx/patches-5.15/950-0198-drm-fourcc-Add-packed-10bit-YUV-4-2-0-format.patch:
```diff
@@ -14,21 +14,9 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
- 		  .vsub = 0, .is_yuv = true },
-+		{ .format = DRM_FORMAT_P030,            .depth = 0,  .num_planes = 2,
-+		  .char_per_block = { 4, 4, 0 }, .block_w = { 3, 0, 0 }, .block_h = { 1, 0, 0 },
-+		  .hsub = 2, .vsub = 2, .is_yuv = true},
```
I think this complete patch was backported, see https://github.com/gregkh/linux/commit/ec1727f89ecd6f2252c0c75e200058819f7ce47a

graysky2 replied:
Removed completely in https://github.com/openwrt/openwrt/commit/9b7d9a03a5872c39d1941f63872cbfeca9c9189d

hauke commented on target/linux/generic/pending-5.15/101-Use-stddefs.h-instead-of-compiler.h.patch:
```diff
@@ -1,22 +0,0 @@
- 
- #include <linux/types.h>
--#include <linux/compiler.h>
-+#include <linux/stddef.h>
```
This was added to stable here: https://github.com/gregkh/linux/commit/c160505c9b574b346031fdf2c649d19e7939ca11

graysky2 replied:
What action do you propose for `ramips/patches-5.15/100-PCI-mt7621-Add-MediaTek-MT7621-PCIe-host-controller-.patch`?  I believe I am not including the `+#include <linux/stddef.h>` you called out as getting added to stable.

hauke replied:
I think we should still remove `drivers/staging/mt7621-pci/pci-mt7621.c` in the `100-PCI-mt7621-Add-MediaTek-MT7621-PCIe-host-controller-.patch`, but we should also backport this patch https://github.com/torvalds/linux/commit/19098934f910b4d47cb30251dd39ffa57bef9523 as a separate patch for example in `target/linux/ramips/patches-5.15/107-...`

This patch is in upstream kernel, but it was backported to the old staging driver in kernel 5.15.

graysky2 replied:
Added and refreshed.  https://github.com/openwrt/openwrt/commit/31a8af0fd62ed2b0168c3674e45adeff291c3a51

graysky2 commented:
Anyone have some insights into the build failure of [Build Kernel with external toolchain (oxnas/ox820) / Build oxnas/ox820](https://github.com/openwrt/openwrt/actions/runs/3813840102/jobs/6487936723#logs)

robimarko committed:
oxnas: sata_oxnas: use ata_link_err

Kernel 5.15.86 has backported ("ata: libata: move ata_{port,link,dev}_dbg
to standard pr_XXX() macros") and this is now causing compilation errors
for oxnas SATA driver due to usage of ata_link_printk().

Upstream has migrated to using the appropriate
ata_link_{err, warn, notice, info} calls a while ago so its not affected.

Lets do the same for oxnas SATA driver and use ata_link_err() instead of
ata_link_printk().

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented:
@graysky2 Found out why it's now breaking:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/include/linux/libata.h?h=v5.15.86&id=3483c3fb4877977d16e547860aa3a1a02a249f81

Basically, upstream moved to using the appropriate ata_link_{err, warn, notice, info} a while ago but we never updated.

This patch will fix compilation, so it would be best to include it as part of this PR:
```
From 9988fbedca3832012aef27856aea2aafd6fbf24a Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Sun, 1 Jan 2023 14:24:06 +0100
Subject: [PATCH] oxnas: sata_oxnas: use ata_link_err

Kernel 5.15.86 has backported ("ata: libata: move ata_{port,link,dev}_dbg
to standard pr_XXX() macros") and this is now causing compilation errors
for oxnas SATA driver due to usage of ata_link_printk().

Upstream has migrated to using the appropriate
ata_link_{err, warn, notice, info} calls a while ago so its not affected.

Lets do the same for oxnas SATA driver and use ata_link_err() instead of
ata_link_printk().

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 target/linux/oxnas/files/drivers/ata/sata_oxnas.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/target/linux/oxnas/files/drivers/ata/sata_oxnas.c b/target/linux/oxnas/files/drivers/ata/sata_oxnas.c
index 217c258565..bd39ce7b1b 100644
--- a/target/linux/oxnas/files/drivers/ata/sata_oxnas.c
+++ b/target/linux/oxnas/files/drivers/ata/sata_oxnas.c
@@ -2039,7 +2039,7 @@ static int sata_oxnas_softreset(struct ata_link *link, unsigned int *class,
 
 	/* if link is occupied, -ENODEV too is an error */
 	if (rc && (rc != -ENODEV || sata_scr_valid(link))) {
-		ata_link_printk(link, KERN_ERR, "SRST failed (errno=%d)\n", rc);
+		ata_link_err(link, "SRST failed (errno=%d)\n", rc);
 		return rc;
 	}
 
-- 
2.38.1
```

graysky2 commented:
@robimarko - I will include it.  Can you suggest the appropriate directory tree in which to place it?

robimarko commented:
@graysky2 Its patch against OpenWrt as the driver is prevent directly in OpenWrt tree, so you just apply the patch with git-am to your tree and push

graysky2 commented:
Got it, thanks.

Borromini commented:
Build system: x86_64
Build-tested: mvebu/cortexa72
Run-tested: mvebu/cortexa72 (RB5009UG+S+IN)
